ast/src/analyzed/display.rs

```diff
@@ -438,6 +455,50 @@ impl Display for PolynomialReference {
     }
 }
 
+impl<E: Display> Display for TraitImplementation<E> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        let type_vars = self
+            .type_scheme
+            .as_ref()
+            .map_or_else(Default::default, |scheme| {
+                if scheme.vars.is_empty() {
+                    Default::default()
+                } else {
+                    format!("<{}>", scheme.vars)
+                }
+            });
+        let trait_vars = self
+            .type_scheme
+            .as_ref()
+            .map_or_else(Default::default, |scheme| {
+                if scheme.types.is_empty() {
+                    Default::default()
+                } else {
+                    let formatted_elements: Vec<String> =
+                        scheme.types.iter().map(|t| format!("{t}")).collect();
+                    format!("<{}>", formatted_elements.join(", "))
+                }
+            });
+
+        let s = format!(
+            "impl{type_vars} {trait_name}{trait_vars} {{\n{methods}}}",
```

Comment:
There is the function `format_type_scheme_around_name` that could help here.

Reply:
I don't think it will work, because trait_vars is a vector (and type_scheme has only one element).

This difference is the result of this separation between TypeScheme and TraitScheme:
https://github.com/powdr-labs/powdr/blob/e904efd693fccc3de3de7f40db8c438fe1e4e3e6/ast/src/parsed/types.rs#L355

Both could be unified if I model the current TypeScheme with a single element vector (ty), but I felt it would add a lot of code changes. What do you think?